pkg/k8sutils/job_test.go

```diff
@@ -817,3 +867,188 @@ func createK8sSecretObj(name string, namespace string, data map[string][]byte) *
+	require.NoError(t, err)
+
+	var eventAsInterfaceWithGitCommitID map[string]interface{}
+	data, err := ioutil.ReadFile("../../test/events/test.triggered.with-gitcommitid.json")
+	require.NoError(t, err)
+
+	err = json.Unmarshal(data, &eventAsInterfaceWithGitCommitID)
+	require.NoError(t, err)
+
+	namespace := testNamespace
+
+	tests := []struct {
+		name               string
+		actionName         string
+		taskName           string
+		expectedActionName string
+		expectedTaskName   string
+		event              map[string]interface{}
+	}{
+		{
+			name:               "Test normal Event with gitcommitid",
+			actionName:         "some_action",
+			taskName:           "task",
+			event:              eventAsInterfaceWithGitCommitID,
+			expectedActionName: "some_action",
+			expectedTaskName:   "task",
+		},
+		{
+			name:               "Test normal Event without gitcommitid",
+			actionName:         "some_action",
+			taskName:           "task",
+			event:              eventAsInterfaceWithoutGitCommitID,
+			expectedActionName: "some_action",
+			expectedTaskName:   "task",
+		},
+		{
+			name:               "Test non k8s compatible action name",
+			actionName:         "Some fancy action name ...",
+			taskName:           "--NameThat's not Compatible with k8s []{}///// a....",
+			event:              eventAsInterfaceWithGitCommitID,
+			expectedActionName: "Some_fancy_action_name",
+			expectedTaskName:   "NameThat_s_not_Compatible_with_k8s_a",
+		},
+	}
+	for i, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+
+			jobName := "some-job-name-" + strconv.Itoa(i)
+			err = k8s.CreateK8sJob(
+				jobName,
+				JobDetails{
+					Action: &config.Action{
+						Name: test.actionName,
+					},
+					Task: &config.Task{
+						Name:  test.taskName,
+						Image: "alpine",
+						Cmd:   []string{"ls"},
+					},
+					ActionIndex:   0,
+					TaskIndex:     0,
+					JobConfigHash: "",
+				},
+				&eventData,
+				JobSettings{
+					JobNamespace: namespace,
+					DefaultResourceRequirements: &corev1.ResourceRequirements{
+						Limits:   make(corev1.ResourceList),
+						Requests: make(corev1.ResourceList),
+					},
+					DefaultPodSecurityContext: new(corev1.PodSecurityContext),
+					DefaultSecurityContext:    new(corev1.SecurityContext),
+				},
+				test.event,
+				namespace,
+			)
+			require.NoError(t, err)
+
+			job, err := k8sClientSet.BatchV1().Jobs(namespace).Get(context.TODO(), jobName, metav1.GetOptions{})
+			require.NoError(t, err)
+
+			expectedLabels := map[string]string{
+				"app.kubernetes.io/managed-by": "job-executor-service",
+				"keptn.sh/context":             test.event["shkeptncontext"].(string),
+				"keptn.sh/ceid":                test.event["id"].(string),
+				"keptn.sh/commitid":            "",
+				"keptn.sh/jes-action":          test.expectedActionName,
+				"keptn.sh/jes-task":            test.expectedTaskName,
+				"keptn.sh/job-confighash":      "",
```

Comment:
just for consistency: shouldn't this label start with `keptn.sh/jes-*`?

```suggestion
				"keptn.sh/jes-job-confighash":      "",
```